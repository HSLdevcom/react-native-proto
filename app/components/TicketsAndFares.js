/**
 * TicketsAndFares component
 * @flow
 */

import React from 'react';
import CustomWebView from './CustomWebView';

function TicketsAndFares() {
    return <CustomWebView autoHeightEnabled onMessageEnabled scrollEnabled={false} uri="https://www.hsl.fi/liput-ja-hinnat?content-only" />;
}

export default TicketsAndFares;
