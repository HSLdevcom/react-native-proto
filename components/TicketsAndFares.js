/**
 * TicketsAndFares component
 * @flow
 */

import React from 'react';
import CustomWebView from './CustomWebView';

function TicketsAndFares() {
    return <CustomWebView uri="https://www.hsl.fi/liput-ja-hinnat" />;
}

export default TicketsAndFares;
