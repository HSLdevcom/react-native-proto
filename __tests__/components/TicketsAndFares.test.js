// https://facebook.github.io/jest/blog/2016/07/27/jest-14.html
import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import TicketsAndFares from '../../components/TicketsAndFares';

// https://github.com/facebook/react-native/issues/12440#issuecomment-282184173
jest.unmock('ScrollView');

it('renders correctly', () => {
    const tree = renderer.create(<TicketsAndFares />).toJSON();
    expect(tree).toMatchSnapshot();
});
