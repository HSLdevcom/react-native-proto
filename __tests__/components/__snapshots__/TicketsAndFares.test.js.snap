// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(0, 122, 201)",
        "borderBottomWidth": 1,
        "borderColor": "rgb(0, 122, 201)",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 50,
      },
      Object {
        "height": 2254,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 8,
        },
        Object {
          "height": 80,
          "left": 335,
          "position": "absolute",
          "width": 80,
          "zIndex": 2,
        },
        null,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={undefined}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript="
                // Workaround to https://github.com/facebook/react-native/issues/10865
                var originalPostMessage = window.postMessage;
                var patchedPostMessage = function(message, targetOrigin, transfer) {
                    originalPostMessage(message, targetOrigin, transfer);
                };
                patchedPostMessage.toString = function() {
                    return String(Object.hasOwnProperty).replace('hasOwnProperty', 'postMessage');
                };
                window.postMessage = patchedPostMessage;
                // window.postMessage(document.cookie);
                // One way to try to send page height back to the app (not working with Android...)
                // https://github.com/scazzy/react-native-webview-autoheight/blob/master/index.js
                // (function(){
                //     let height = 0;
                //     if(document.documentElement.clientHeight>document.body.clientHeight) {
                //         height = document.documentElement.clientHeight;
                //     } else {
                //         height = document.body.clientHeight;
                //     }
                //     postMessage('height;' + height);
                // })();
            "
      mediaPlaybackRequiresUserAction={undefined}
      messagingEnabled={true}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={undefined}
      scrollEnabled={false}
      source={
        Object {
          "uri": "https://www.hsl.fi/liput-ja-hinnat?content-only",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "marginTop": 63,
            },
            Object {
              "marginTop": 0,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;
